pkg/reconciler/ksvc/resources/knative_configuration.go

```diff
@@ -2,77 +2,72 @@ package resources
 
 import (
 	"fmt"
-
 	"github.com/knative/serving/pkg/apis/autoscaling"
 	knservingv1alpha1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/knative/serving/pkg/apis/serving/v1beta1"
 	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha1"
 	"github.com/kubeflow/kfserving/pkg/constants"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func CreateKnativeConfiguration(kfsvc *v1alpha1.KFService) (*knservingv1alpha1.Configuration, *knservingv1alpha1.Configuration) {
+func CreateKnativeConfiguration(name string, metadata metav1.ObjectMeta, modelSpec *v1alpha1.ModelSpec) *knservingv1alpha1.Configuration {
+	if modelSpec == nil {
+		return nil
+	}
 	annotations := make(map[string]string)
-	if kfsvc.Spec.MinReplicas != 0 {
-		annotations[autoscaling.MinScaleAnnotationKey] = fmt.Sprint(kfsvc.Spec.MinReplicas)
+	if modelSpec.MinReplicas != 0 {
+		annotations[autoscaling.MinScaleAnnotationKey] = fmt.Sprint(modelSpec.MinReplicas)
 	}
-	if kfsvc.Spec.MaxReplicas != 0 {
-		annotations[autoscaling.MaxScaleAnnotationKey] = fmt.Sprint(kfsvc.Spec.MaxReplicas)
+	if modelSpec.MaxReplicas != 0 {
+		annotations[autoscaling.MaxScaleAnnotationKey] = fmt.Sprint(modelSpec.MaxReplicas)
 	}
 
-	defaultConfiguration := &knservingv1alpha1.Configuration{
+	// User can pass down scaling target annotation to overwrite the target default 1
+	if _, ok := metadata.Annotations[autoscaling.TargetAnnotationKey]; !ok {
+		annotations[autoscaling.TargetAnnotationKey] = constants.DefaultScalingTarget
+	}
+	// User can pass down scaling class annotation to overwrite the default scaling KPA
+	if _, ok := metadata.Annotations[autoscaling.ClassAnnotationKey]; !ok {
+		annotations[autoscaling.ClassAnnotationKey] = autoscaling.KPA
+	}
+
+	kfsvcAnnotations := filter(metadata.Annotations, configurationAnnotationFilter)
+
+	configuration := &knservingv1alpha1.Configuration{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        constants.DefaultConfigurationName(kfsvc.Name),
-			Namespace:   kfsvc.Namespace,
-			Labels:      kfsvc.Labels,
-			Annotations: union(kfsvc.Annotations, annotations),
+			Name:      name,
+			Namespace: metadata.Namespace,
+			Labels:    metadata.Labels,
```

Comment:
Why not ignore Annotations here?

Reply:
Annotations need to be set on revision template not on configuration's metadata, otherwise it won't propagate down.
https://github.com/knative/serving/blob/release-0.5/pkg/reconciler/v1alpha1/configuration/resources/revision.go#L32

Reply:
make sense, thanks